src/pathfinding/grid_pathfinder.jl

```diff
@@ -0,0 +1,362 @@
+export CostMetric,
+    DirectDistance,
+    Chebyshev,
+    HeightMap,
+    AStar,
+    delta_cost,
+    find_path,
+    set_target!,
+    is_stationary,
+    heightmap,
+    walkmap
+
+"""
+    Path{D}
+Alias of `MutableLinkedList{Dims{D}}`. Used to represent the path to be
+taken by an agent in a `D` dimensional [`GridSpace{D}`](@ref).
+"""
+const Path{D} = MutableLinkedList{Dims{D}}
+
+"""
+    CostMetric{D}
+An abstract type representing a metric that measures the approximate cost of travelling
+between two points in a `D` dimensional [`GridSpace{D}`](@ref).
+"""
+abstract type CostMetric{D} end
+
+struct DirectDistance{D} <: CostMetric{D}
+    direction_costs::Vector{Int}
+end
+
+Base.show(io::IO, metric::DirectDistance) = print(io, "DirectDistance")
+
+"""
+    DirectDistance
```

Comment:
```suggestion
    DirectDistance{D} <: CostMetric{D}
```

Reply:
Futhermore the input must be stated: `DirectDistance{D}([argument])`

Reply:
I left out the `<: CostMetric{D}` since, for example, `GridSpace{D,P,W} <: DiscreteSpace` isn't in the docstring. I'll add the direction costs as an optional argument, in `[...]`